src/main/java/org/broadinstitute/barclay/help/HelpDoclet.java

```diff
@@ -307,35 +306,41 @@ private DocumentedFeatureObject getFeatureForClass(final Class<?> clazz) {
             String summary = f.summary();
             String groupName = f.groupName();
             String groupSummary = f.groupSummary();
+
             CommandLineProgramProperties clProps =
                     clazz.isAnnotationPresent(CommandLineProgramProperties.class) ?
                         clazz.getAnnotation(CommandLineProgramProperties.class) :
                         null;
-            if ((summary == null || summary.length() == 0) && clProps != null) {
-                summary = clProps.oneLineSummary();
-                if (summary == null || summary.length() == 0) {
-                    summary = "No summary available";
-                }
-            }
-            if ((groupName == null || groupName.length() == 0) && clProps != null) {
-                groupName = clProps.programGroup().getName();
-                if (groupName == null || groupName.length() == 0) {
-                    groupName = "No group name available";
-                }
-            }
-            if ((groupSummary == null || groupSummary.length() == 0) && clProps != null) {
+            CommandLineProgramGroup clpGroup = null;
+            if (clProps != null) {
                 try {
-                    groupSummary = clProps.programGroup().newInstance().getDescription();
-                    if (groupSummary == null || groupSummary.length() == 0) {
-                        groupSummary = "No group summary available";
-                    }
-                }
-                catch (IllegalAccessException | InstantiationException e){
+                    clpGroup = clProps.programGroup().newInstance();
+                } catch (IllegalAccessException | InstantiationException e) {
                     logger.warn(
                             String.format("Cant instantiate program group class to retrieve summary for group %s for class %s",
                                     clProps.programGroup().getName(),
                                     clazz.getName()));
-                    groupSummary = "No group summary available";
+                }
+            }
+            if (summary.length() == 0) {
+                if (clpGroup != null) {
+                    summary = clProps.oneLineSummary();
+                }
+                if (summary.length() == 0) {
+                    logger.warn("No summary declared for: " + clazz.getCanonicalName() + ". Trying javadoc fallback.");
```

Comment:
They're never null because the annotation attributes now all have default values of "". We no longer want to populate the strings with "No X declared" here since further on, when we have access to the javadoc, we'll use the empty string to determine if we should attempt to fall back to get a summary from that. I'll add a comment explaining that here. Ultimately this all needs refactoring.